docs/develop/python/user-metadata.mdx

```diff
@@ -0,0 +1,259 @@
+    async def run(self, input: str) -> str:
+        # Get the current details
+        current_details = workflow.get_current_details()
+        print(f"Current details: {current_details}")
+        
+        # Set/update the current details
+        workflow.set_current_details("Updated workflow details with new status")
+        
+        return "Workflow completed"
+```
+
+### Activity
+
+You can attach a summary to activities when starting them from within a workflow:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Start an activity with a summary
+        result = await workflow.execute_activity(
+            your_activity,
+            input,
+            start_to_close_timeout=timedelta(seconds=10),
+            summary="Processing user data"
+        )
+        return result
+```
+
+### Timer
+
+You can attach a summary to timers within a workflow:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Create a timer with a summary
+        await workflow.sleep(timedelta(minutes=5), summary="Waiting for payment confirmation")
+        return "Timer completed"
+```
+
+### Signal Handler
+
+When defining a signal handler in a workflow, you can include a description:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    def __init__(self):
+        self._approved = False
+
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Wait for approval signal
+        await workflow.wait_condition(lambda: self._approved)
+        return "Workflow approved"
+
+    @workflow.signal(description="Signal to approve the workflow")
+    def approve(self) -> None:
+        self._approved = True
+```
+
+### Query Handler
+
+When defining a query handler in a workflow, you can include a description:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    def __init__(self):
+        self._status = "pending"
+
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Workflow logic
+        self._status = "processing"
+        await workflow.sleep(timedelta(minutes=1))
+        self._status = "completed"
+        return "Workflow completed"
+
+    @workflow.query(description="Get the current status of the workflow")
+    def get_status(self) -> str:
+        return self._status
+```
+
+### Update Handler
+
+When defining an update handler in a workflow, you can include a description:
+
+```python
+@workflow.defn
+class YourWorkflow:
+    def __init__(self):
+        self._shipping_address = ""
+
+    @workflow.run
+    async def run(self, input: str) -> str:
+        # Workflow logic
+        return f"Order shipped to: {self._shipping_address}"
+
+    @workflow.update(description="Update the shipping address for the order")
+    def change_shipping_address(self, new_address: str) -> None:
+        self._shipping_address = new_address
+```
+
+## Retrieving Metadata
```

Comment:
I don't think retrieving metadata really needs to be documented tbh. This is mostly only done by Temporal interfaces, the user need to retrieve this information is almost non-existent. (could be argued the static summary/details stuff has value, but really it's just a part of everything else you can describe, doesn't deserve special docs)